ckanext/restricted/tests/test_plugin.py

```diff
@@ -282,6 +282,22 @@ def test_user_can_see_only_accessible_resource_in_package_search(self):
         assert len(package['resources']) == 1
         assert package['resources'][0]['id'] == other_resource['id']
 
+    @pytest.mark.ckan_config(u'ckan.auth.allow_dataset_collaborators', 'true')
+    def test_collaborator_overrides_restricted_settings(self):
+        dataset, other, other_resource, org_resource = self._two_users_one_package_two_resources_one_restricted()
+        # when:
+        context = {
+            'ignore_auth': False,
+            'user': other['name']
+        }
+        with pytest.raises(logic.NotAuthorized):
+            logic.check_access('resource_show', context, {'id': org_resource['id']})
```

Comment:
This check should not be here. We already assume that core restricted functionality is working ok and that "other" user does not have access to "org_resource".
This test shouldn't check this. The real "when" part is below at the `helpers.call_action`.

This test should fail only after access is denied for dataset with a collaborator. It shouldn't fail when restricted_resource_show is broken.